vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/test/java/com/vaadin/flow/component/sidenav/tests/SideNavItemTest.java

```diff
@@ -475,6 +478,186 @@ public void setPathFromComponentAndRouteParameters_pathContainsParameters() {
         });
     }
 
+    @Test
+    public void createFromComponentWithRouteParameters_aliasContainsParameters() {
+        runWithMockRouter(TestRouteWithAliases.class, () -> {
+            SideNavItem item = new SideNavItem("test",
+                    TestRouteWithAliases.class,
+                    new RouteParameters(Map.of("key", "value")));
+
+            Assert.assertTrue(item.getPathAliases().contains("foo/value/bar"));
+        });
+    }
+
+    @Test
+    public void setPathAndRouteParametersAsComponent_aliasContainsParameters() {
+        runWithMockRouter(TestRouteWithAliases.class, () -> {
+            SideNavItem item = new SideNavItem("test");
+            item.setPath(TestRouteWithAliases.class,
+                    new RouteParameters(Map.of("key", "value")));
+
+            Assert.assertTrue(item.getPathAliases().contains("foo/value/bar"));
+        });
+    }
+
+    @Test
+    public void createWithNoPathAlias_pathAliasesEmpty() {
+        final SideNavItem item = new SideNavItem("Test");
+
+        Assert.assertEquals(0, item.getPathAliases().size());
+        Assert.assertNull(item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setEmptyPathAlias_pathAliasAdded() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases("");
+
+        Assert.assertEquals(Set.of(""), item.getPathAliases());
+        Assert.assertEquals("[\"\"]",
+                item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setMultiplePathAliases_pathAliasesAdded() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases("alias1", "alias2");
+
+        Assert.assertEquals(Set.of("alias1", "alias2"), item.getPathAliases());
+        Assert.assertTrue(Set
+                .of("[\"alias1\",\"alias2\"]", "[\"alias2\",\"alias1\"]")
+                .contains(item.getElement().getProperty("pathAliases")));
+    }
+
+    @Test
+    public void setPathAlias_removePathAlias_pathAliasesEmpty() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases("alias");
+        item.removePathAliases("alias");
+
+        Assert.assertEquals(0, item.getPathAliases().size());
+        Assert.assertNull(item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setMultiplePathAliases_clearPathAliases_pathAliasesEmpty() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases("alias1", "alias2");
+        item.clearPathAliases();
+
+        Assert.assertEquals(0, item.getPathAliases().size());
+        Assert.assertNull(item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setPathAsComponent_setNullAsComponent_pathAliasesEmpty() {
```

Comment:
Could also add a test that sets a different component and checks that aliases are replaced.

Reply:
Added 2 tests for this, will refactor them into a single test with the test refactor PR.